runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -1306,22 +1317,24 @@ bool
 ROMClassBuilder::compareROMClassForEquality(U_8 *romClass,bool romClassIsShared,
 		ROMClassWriter *romClassWriter, SRPOffsetTable *srpOffsetTable, SRPKeyProducer *srpKeyProducer,
 		ClassFileOracle *classFileOracle, U_32 modifiers, U_32 extraModifiers, U_32 optionalFlags,
-		ROMClassCreationContext * context, U_32 romSize, bool isLambda)
+		ROMClassCreationContext * context, U_32 sizeToCompareForLambda, bool isLambda)
 {
 	bool ret = false;
 
 	if (isLambda) {
-		if (sizeof(U_64) < abs((int)(romSize - ((J9ROMClass *)romClass)->romSize))) {
-			/* If the class is a lambda class, we compare the romSizes first to save time. Lambda class names are in the format of
-			 * HostClassName$$Lambda$<IndexNumber>/0x0000000000000000. When we reach this check, the host class names will be the
-			 * same for both the classes because of the hash key check earlier so the only difference in the size will be the
-			 * difference between the number of digits of the index number. The same lambda class might have a different index
-			 * number from run to run and when the number of digits of the index number increases by 1, romSize increases by 2.
-			 * We check if the difference between romSizes is bigger than sizeof(U_64) because this will allow at least 4 but up
-			 * to 7 (because romSize has padding and is always multiples of 8) digits difference. (eg. HostClassName$$Lambda$[1-9]
-			 * can get matched at least up to HostClassName$$Lambda$99999, HostClassName$$Lambda$[10-99] can get matched at least
-			 * up to HostClassName$$Lambda$999999) This check is different than the classFileSize check because when the number of
-			 * digits of the index number increases by 1, romSize increases by 2 but classFileSize increases by 1.
+		if (sizeof(U_32) < abs((int)(sizeToCompareForLambda - ((J9ROMClass *)romClass)->classFileSize))) {
```

Comment:
Why did this change to U32 from U64? Also, the use of abs seems questionable (I realize it was there before).

Reply:
I removed the original check on romSize and move the check of classFileSize from ComparingCursor::writeU32() to here (without changing the classFileSize check itself).

I think the reason of using abs() is because the `<indexNumber>` in the lambda class name: `HostClassName$$Lambda$<IndexNumber>/0x0000000000000000`, which is generated at runtime. 
For the same lambda class, IndexNumber could be for example, a single digit in run 1 and a double-digit in run 2. It could also happen the opposite way. The comment says "the number of digits of the index number increases by 1 .... classFileSize increases by 1". So abs() is used.

Reply:
Can't the index number vary hugely? This still doesn't explain why the U32 or U64 size check is there. Perhaps this should be the actual variance value (i.e. the difference between the smallest and larger printed index number). It seems to have no relation to the size of a C type.

Reply:
> It seems to have no relation to the size of a C type.

I agree. 

The indexNumber is counter for the number of lambda classes defined so far. It is an int in the JCL side. So theory, it could vary from 0 to 2147483647. So less than 10 bytes.

Reply:
Changed to allow variance from 0 to 9 bytes .

Reply:
Please update with that - also, please flip the comparison (`constant < computed` is much more difficult to reason than `computed >= constant`)